### DISCUSSION TITLE
Simulate the behavior of pip's `--extra-index-url`

### DISCUSSION THREAD

**Discussion by bruno-brant** [python-poetry/poetry#6401] [Q&A]
I'm trying to install a CUDA distribution of pytorch, and [the documentation](https://pytorch.org/get-started/locally/) asks me to use a extra index:

``` bash
pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116
```

I tried simulating this behavior by adding a additional source to Poetry: 

```bash
poetry source add cu116 https://download.pytorch.org/whl/cu116 --secondary
```

However, Poetry began to try to download everything from there, instead of just `torch` and `torchvision`. 

Is there a way to replicate this behavior?

**Comment by neersighted** [comment]
This is currently the intended behavior of secondary sources. They were mostly an afterthought not originally intended for the 'specific packages from this source use case.' We are actively discussing new designs around repository searching, and how to move forward in a backwards compatible way.
 
Please see my proposal at https://github.com/python-poetry/poetry/pull/5984#issuecomment-1237245571 and offer any input you have there -- we will likely migrate discussion to a new PR or issue if the original author does not want to see an agreed upon design through, but for now that is the correct venue.

**Reply by neersighted** [reply]
I suppose it's also worth noting that a non-secondary source is actually the equivalent to `--extra-index-url` -- pip just doesn't say anything when it tries an index for a package and fails, and is much faster than us as they do not gather package metadata in their resolver.

**Comment by bruno-brant** [comment]
Hello @neersighted, thanks for the response. I'm not that versed in pip, good to know know. That means if I set it up correctly I can get the behavior I want.

I'll try to contribute to the proposal. I think Poetry is a must in the Python ecosystem. I migrated a number of internal projects in my org from stupid requirements.txt to it.
